tests/integration/main.rs

```diff
@@ -0,0 +1,197 @@
+
+use pretty_assertions::assert_eq;
+
+use arrow::array::StructArray;
+use arrow::record_batch::RecordBatch;
+use datafusion_orc::arrow_reader::ArrowReaderBuilder;
+
+/// Checks parsing a `.orc` file produces the expected result in the `.jsn.gz` path
+fn test_expected_file(name: &str) {
+    let dir = env!("CARGO_MANIFEST_DIR");
+    let orc_path = format!("{}/tests/integration/data/{}.orc", dir, name);
+    let jsn_gz_path = format!("{}/tests/integration/data/expected/{}.jsn.gz", dir, name);
+    let f = File::open(orc_path).expect("Could not open .orc");
+    let builder = ArrowReaderBuilder::try_new(f).unwrap();
+    let orc_reader = builder.build();
+    let total_row_count = orc_reader.total_row_count();
+
+    // Read .orc into JSON objects
+    let batches: Vec<RecordBatch> = orc_reader.collect::<Result<Vec<_>, _>>().unwrap();
+    let objects: Vec<serde_json::Value> = batches
+        .into_iter()
+        .map(|batch| -> StructArray { batch.into() })
+        .flat_map(|array| {
+            arrow_json::writer::array_to_json_array(&array)
+                .expect("Could not serialize convert row from .orc to JSON value")
+        })
+        .collect();
+
+    // Read expected JSON objects
+    let mut expected_json = String::new();
+    flate2::read::GzDecoder::new(&File::open(jsn_gz_path).expect("Could not open .jsn.gz"))
+        .read_to_string(&mut expected_json)
+        .expect("Could not read .jsn.gz");
+
+    let objects_count = objects.len();
+
+    // Reencode the input to normalize it
+    let expected_lines = expected_json
+        .split('\n')
+        .filter(|line| !line.is_empty())
+        .map(|line| {
+            serde_json::from_str::<serde_json::Value>(line)
+                .expect("Could not parse line in .jsn.gz")
+        })
+        .map(|v| {
+            serde_json::to_string_pretty(&v).expect("Could not re-serialize line from .jsn.gz")
+        })
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    let lines = objects
+        .into_iter()
+        .map(|v| serde_json::to_string_pretty(&v).expect("Could not serialize row from .orc"))
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    if lines.len() < 1000 {
+        assert_eq!(lines, expected_lines);
+    } else {
+        // pretty_assertions consumes too much RAM and CPU on large diffs,
+        // and it's unreadable anyway
+        assert_eq!(lines[0..1000], expected_lines[0..1000]);
+        assert!(lines == expected_lines);
+    }
+
+    assert_eq!(total_row_count, objects_count as u64);
+}
+
+#[test]
+fn columnProjection() {
+    test_expected_file("TestOrcFile.columnProjection");
+}
+#[test]
+fn emptyFile() {
+    test_expected_file("TestOrcFile.emptyFile");
+}
+#[test]
+#[ignore] // Why?
+fn metaData() {
+    test_expected_file("TestOrcFile.metaData");
+}
+#[test]
+#[ignore] // Why?
+fn test1() {
+    test_expected_file("TestOrcFile.test1");
+}
+#[test]
+#[should_panic] // Incorrect timezone + representation differs
```

Comment:
Thanks for leaving these annotations, gives us a goal to work towards

For all tests that fail we can just `#[ignore]` them and leave TODO comments to address these